modules/videoeditor-ffmpeg-impl/src/main/java/org/opencastproject/videoeditor/subtitle/webvtt/WebVTTParser.java

```diff
@@ -0,0 +1,189 @@
+/**
+ * Licensed to The Apereo Foundation under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ *
+ * The Apereo Foundation licenses this file to you under the Educational
+ * Community License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License
+ * at:
+ *
+ *   http://opensource.org/licenses/ecl2.txt
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ */
+package org.opencastproject.videoeditor.subtitle.webvtt;
+
+import org.opencastproject.videoeditor.subtitle.base.SubtitleParsingException;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Parses WebVTT from a file into a datastructure to allow for easy modification.
+ * Throws exceptions if the read WebVTT is invalid.
+ *
+ * TODO: Comments are currently ignored and discarded. Find a good way to keep comments
+ *  without compromising easy editing.
+ */
+public class WebVTTParser {
+
+  private static final String WEBVTT_METADATA_HEADER_STRING = "\\S*[:=]\\S*";
+  private static final Pattern WEBVTT_METADATA_HEADER =
+          Pattern.compile(WEBVTT_METADATA_HEADER_STRING);
+
+  // Regex checks if not a time interval
+  private static final String WEBVTT_CUE_IDENTIFIER_STRING = "^(?!.*(-->)).*$";
+  private static final Pattern WEBVTT_CUE_IDENTIFIER =
+          Pattern.compile(WEBVTT_CUE_IDENTIFIER_STRING);
+
+  // Timestamp from time interval
+  private static final String WEBVTT_TIMESTAMP_STRING = "(\\d+:)?[0-5]\\d:[0-5]\\d\\.\\d{3}";
+  private static final Pattern WEBVTT_TIMESTAMP = Pattern.compile(WEBVTT_TIMESTAMP_STRING);
+
+  private Charset charset; // Charset of the input files
+
+  public WebVTTParser() {
+    this.charset = StandardCharsets.UTF_8;
+  }
+
+  public WebVTTParser(Charset charset) {
+    this.charset = charset;
+  }
+
+  public WebVTTSubtitle parse(InputStream is) throws IOException, SubtitleParsingException {
+    return parse(is, true);
+  }
+
+  public WebVTTSubtitle parse(InputStream is, boolean strict) throws IOException, SubtitleParsingException {
```

Comment:
The parameter `strict` isn't actually used anywhere, is it?

```suggestion
  public WebVTTSubtitle parse(InputStream is) throws IOException, SubtitleParsingException {
```